TICC.ino

```diff
@@ -7,52 +7,32 @@
 // Portions Copyright George Byrkit K9TRV 2016
 // Licensed under BSD 2-clause license
 
+
+#include "TICC.h"     //Register and structure definitions
+
 // the TI TDC7200 communicates using SPI, so include the library:
 #include <SPI.h>
 
-// set up for ISR
-int interrupt = 1; // interrupt for Uno pin 3
-int int_pin = 3; // physical pin
-volatile unsigned long long coarse_count = 0;
-
-// hardware connections to TDC2700 -- need to reassign these more logically (todo in Eagle)
-const int ENABLE_A = 13;
-const int INTB_A = 12;
-const int CSB_A = 11;
-const int ENABLE_B = 10;
-const int INTB_B = 9;
-const int CSB_B = 8;
-const int STOP_A = 7;
-const int STOP_B = 8;
-
-const int ChA = 0;	// channel A
-const int ChB = 1;	// channel B
-
-// TDC7200 register addresses
-const int CONFIG1 = 0x00;                 // default 0x00
-const int CONFIG2 = 0x01;                 // default 0x40
-const int INT_STATUS = 0x02;              // default 0x00
-const int INT_MASK = 0x03;                // default 0x07
-const int COARSE_CNTR_OVF_H = 0x04;       // default 0xff
-const int COARSE_CNTR_OVF_L = 0x05;       // default 0xff
-const int CLOCK_CNTR_OVF_H = 0x06;        // default 0xff
-const int CLOCK_CNTR_OVF_L = 0x07;        // default 0xff
-const int CLOCK_CNTR_STOP_MASK_H = 0x08;  // default 0xff
-const int CLOCK_CNTR_STOP_MASK_L = 0x09;  // default 0xff
-// gap from 0x0A thru 0x0F...
-const int TIME1 = 0x10;                   // default 0x00_0000
-const int CLOCK_COUNT1 = 0x11;            // default 0x00_0000
-const int TIME2 = 0x12;                   // default 0x00_0000
-const int CLOCK_COUNT2 = 0x13;            // default 0x00_0000
-const int TIME3 = 0x14;                   // default 0x00_0000
-const int CLOCK_COUNT3 = 0x15;            // default 0x00_0000
-const int TIME4 = 0x16;                   // default 0x00_0000
-const int CLOCK_COUNT4 = 0x17;            // default 0x00_0000
-const int TIME5 = 0x18;                   // default 0x00_0000
-const int CLOCK_COUNT5 = 0x19;            // default 0x00_0000
-const int TIME6 = 0x1A;                   // default 0x00_0000
-const int CALIBRATION1 = 0x1B;            // default 0x00_0000
-const int CALIBRATION2 = 0x1C;            // default 0x00_0000
+volatile unsigned long long coarseCount = 0;
+
+// Enumerate the TDC7200 channel structures
+static chanTDC7200 ChanA = {
+  .ID = 'A',
+  .ENABLE = ENABLE_A,
+  .INTB = INTB_A,
+  .CSB = CSB_A,
+  .STOP = STOP_A };
+
+static chanTDC7200 ChanB = {
+  .ID = 'B',
+  .ENABLE = ENABLE_B,
+  .INTB = INTB_B,
+  .CSB = CSB_B,
+  .STOP = STOP_B };
+
+chanTDC7200* ChA = &ChanA;  // pointer to Channel A
+chanTDC7200* ChB = &ChanB;
+
```

Comment:
Why are you defining variables that are just pointers to global static variables?  This makes it spaghetti-like to trace things back to what they really are.  When you use them in main(), I have to go up here and find them as pointers that I then have to go up and find the static definitions of them.  See more comments below...